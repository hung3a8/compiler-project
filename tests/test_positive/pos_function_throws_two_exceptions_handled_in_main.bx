exception DivByZero;
exception OutOfBounds;

def foo(x : int) : int raises DivByZero, OutOfBounds {
    if (x == 0) {
        raise DivByZero;
    }
    if (x > 100) {
        raise OutOfBounds;
    }
    return x;
}

def main() {
    try {
        var y = foo(0) : int;
    } except DivByZero {
        print(42);
    } except OutOfBounds {
        print(43);
    }
    
    try {
        var z = foo(101) : int;
    } except DivByZero {
        print(44);
    }
    except OutOfBounds {
        print(45);
    }
    
    print(1);
}
