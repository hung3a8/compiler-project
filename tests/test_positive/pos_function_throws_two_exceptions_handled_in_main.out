42
45
1
